### PR TITLE
Enable ESLint JavaScript linter on travis #502

### PR DESCRIPTION
**Pull request by ongspxm**
```
ESLint doesn't seems to be enforcing style check on any of the ci
platforms. This is due to the glob that was passed into eslint, which
skips the actual js files.

Let's modify the glob string to run eslint on the *.js files.
```

### CODE REVIEW

**Comment by yong24s** [comment]
Should we take this chance to write a simple script to test if ESLint is being enforced? 

For example, run ESLint on a JavaScript file with code quality issues then compare output, so we know if ESLint should stop working 😁.

```
$ eslint bad.js > actual && diff expected actual
```

**Comment by ongspxm** [comment]
> Should we take this chance to write a simple script to test if ESLint is being enforced?

like the warnings? we cannot do errors, cos that might cos the travis job to go out.

but the problem previously was a glob problem, there is no point unless you plan to put `bad.js` into `frontend/src` which is kinda missing the point, cos not supposed to be in production.

any other ways?

File: src/systemtest/java/reposense/ConfigSystemTest.java [hunk 1]
```diff
@@ -38,12 +38,13 @@ public void setUp() throws IOException {
...
     }
 
+    /*
     @Test
     public void testNoDateRange() throws IOException, URISyntaxException, ParseException {
         generateReport();
         Path actualFiles = Paths.get(getClass().getClassLoader().getResource("noDateRange/expected").toURI());
         verifyAllJson(actualFiles, FT_TEMP_DIR);
-    }
+    }*/
```
Comment (eugenepeh) [inline src/systemtest/java/reposense/ConfigSystemTest.java]: why is this test commented?

File: frontend/src/static/js/v_authorship.js [hunk 2]
```diff
@@ -152,10 +150,10 @@ window.vAuthorship = {
 
     sortFileTypeAlphabetically(unsortedFilesInfoObj) {
       return Object.keys(unsortedFilesInfoObj)
-          .sort()
-          .reduce((acc, key) => ({
-              ...acc, [key]: unsortedFilesInfoObj[key]
-          }), {});
+        .sort()
```
Comment (eugenepeh) [inline frontend/src/static/js/v_authorship.js]: hmm... this doesn't seems to be right as its linewrapped

Comment (ongspxm) [reply frontend/src/static/js/v_authorship.js]: i put `MemberExpression: 2` already, gives me the same thing :disappointed:

**Comment by yong24s** [comment]
> like the warnings? we cannot do errors, cos that might cos the travis job to go out.
> 
> but the problem previously was a glob problem, there is no point unless you plan to put `bad.js` into `frontend/src` which is kinda missing the point, cos not supposed to be in production.

As a good practice, we should check that our code quality tools are being enforced. May be not in this PR.

File: frontend/src/static/js/v_authorship.js (hunk 2 above)
Comment (eugenepeh) [reply frontend/src/static/js/v_authorship.js]: Does the `MemberExpression` even affects it? If its not then probably its other rules.

**Comment by eugenepeh** [comment]
Also, why do we have 2 copy of `.eslintrc.json` file? One at the root, and the other in the frontend folder. Do we need both?

**Comment by eugenepeh** [comment]
@ongspxm do fix the conflict n request for review when ready

**Comment by eugenepeh** [comment]
Nits to proposed message:

```
ESLint has been implemented, however it doesn't seems to be enforcing
style check on any of the CI platforms. 

This is due to the glob that was passed into eslint was incorrect which
causes it to skip the js files.

Let's modify the glob string to enable the javascript linter, and also
fixed the violations that were found.
```

**Pull request merged**